python/graphstorm/inference/emb_infer.py

```diff
@@ -23,45 +23,53 @@
 
 
 class GSgnnEmbGenInferer(GSInferrer):
-    """ Embedding Generation inferrer.
+    """ Inferrer for embedding generation tasks.
 
-    This is a high-level inferrer wrapper that can be used directly
-    to generate embedding for inferer.
+    ``GSgnnEmbGenInferer`` inherits from the ``GSInferrer`` and use the functions
+    provided by ``GSInferrer`` to define the ``infer()`` method that performs one work:
```

Comment:
These are implementation details right? As the end user we're interested in the public methods, not how they are derived I think. Since this is a user-facing API, we could change where the functions are defined and the API would remain stable, so I don't think we need to mention the inheritance structure.

Reply:
Agree. Updated and only keep the description of this class' fuctionality.

Reply:
Agree. Update all Trainers and Inferrers class doc string to avoid implementation description.